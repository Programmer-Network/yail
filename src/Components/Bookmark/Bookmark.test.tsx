import { fireEvent, render, screen } from "@testing-library/react";
import { vi } from "vitest";

import { Bookmark } from "./";

describe("Bookmark component", () => {
  const args = {
    isBookmarked: false,
    handleDelete: () => null,
    handleAdd: () => null,
    isLoading: false
  };

  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(
      <Bookmark {...args} isBookmarked={true} isLoading={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays spinner when loading", () => {
    render(<Bookmark {...args} isLoading={true} />);
    expect(screen.getByTestId("icon-spinner")).toBeInTheDocument();
  });

  test("displays bookmarked icon when isBookmarked is true", () => {
    render(<Bookmark {...args} isBookmarked={true} />);
    expect(screen.getByTestId("icon-bookmarked")).toBeInTheDocument();
  });

  test("displays bookmark icon when isBookmarked is false", () => {
    render(<Bookmark {...args} isBookmarked={false} />);
    expect(screen.getByTestId("icon-bookmark")).toBeInTheDocument();
  });

  test("calls handleDelete on bookmarked icon click", () => {
    const handleDelete = vi.fn();
    render(
      <Bookmark {...args} isBookmarked={true} handleDelete={handleDelete} />
    );
    fireEvent.click(screen.getByTestId("icon-bookmarked"));
    expect(handleDelete).toHaveBeenCalledTimes(1);
  });

  test("calls handleAdd on bookmark icon click", () => {
    const handleAdd = vi.fn();
    render(<Bookmark {...args} isBookmarked={false} handleAdd={handleAdd} />);
    fireEvent.click(screen.getByTestId("icon-bookmark"));
    expect(handleAdd).toHaveBeenCalledTimes(1);
  });
});
