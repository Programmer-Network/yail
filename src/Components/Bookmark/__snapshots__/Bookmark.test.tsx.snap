// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bookmark component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <div
    class="flex items-start gap-2"
  >
    <div
      class="flex gap-2"
    >
      <div
        class="relative top-[-1px] w-5 cursor-pointer fill-primary hover:fill-primary"
        data-icon="IconBookmarked"
        data-testid="icon-bookmarked"
      >
        [Icon: IconBookmarked]
      </div>
    </div>
  </div>
</DocumentFragment>
`;
